File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/String/28_ImplementstrStr.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-06 21:22:30
 */

c.. Solution {
public:
    int strStr(string haystack, string needle) {
        __(needle__"")  r_ 0;
        int str_len = haystack.size();
        int pat_len = needle.size();
        ___(int i=0;i<str_len-pat_len+1;i++){
            int j=0;
            ___(;j<pat_len;j++){
                __(haystack[i+j] != needle[j]){
                    ______;
                }
            }
            __(j __ pat_len){
                r_ i;
            }
        }
        r_ -1;
    }
};

/*
""
""
"abaa"
"aa"
"aaabbb"
"abbb"
*/

File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/String/28_ImplementstrStr.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-


c.. Solution o..
    # Not notoriously hard-to-understand algorithm KMP
    ___ strStr  haystack, needle
        # Return 0 if needle is ""
        __ n.. needle:
            r_ 0
        length = l..(haystack)

        # If one char in haystack is same with needle[0],
        # then verify the other chars in needle.
        ___ i __ r..(length-l..(needle)+1
            __ haystack[i:i+l..(needle)] __ needle:
                r_ i

        r_ -1

"""
""
""
"abaa"
"aa"
"aaabbb"
"abbb"
"""
